utils/src/main/java/uk/gov/pay/commons/utils/xray/XRaySessionProfiler.java

```diff
@@ -41,21 +47,23 @@ public void endOperationProfile(String operationName, DatabaseQuery databaseQuer
     @Override
     public Object profileExecutionOfQuery(DatabaseQuery databaseQuery, Record record, AbstractSession abstractSession) {
         if (recorder.getCurrentSegmentOptional().isPresent()) {
-            final Optional<Connection> maybeConnection = Optional.ofNullable(abstractSession.getAccessor()).map(Accessor::getConnection);
-            Map<String, Object> databaseMetadata = new HashMap<>();
+            DatabaseLogin databaseLogin = (DatabaseLogin) abstractSession.getDatasourceLogin();
```

Comment:
I see that `getDatasourceLogin()` return `Login` which has three concrete implementations, `DatabaseLogin` which we're casting to here, and also `EISLogin` and `XMLLogin`. All extend abstract class `DatasourceLogin` but that doesn't have `getUrl()` hence why we're casting to `DatabaseLogin`. I can't see any cause for us using EISLogin nor XMLogin so I guess this should be ok (otherwise the cast will fail at runtime if we do use those).